### PR TITLE
[MM-23225] Add ability to get seed from environment variable

### PR DESCRIPTION
**Pull request by isacikgoz**
#### Summary
This PR adds:
- Global seed when running 'loadtest`
- `TestMain` functions to be able to set rand seed. 

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-23225

### CODE REVIEW

**Comment by streamer45** [comment]
What about tests? Do they run with the default seed of `1`?

**Comment by agnivade** [comment]
Is the ability to disable the random seeding really needed though ? If we do use it, we should prefix it with "MM_".

**Comment by streamer45** [comment]
> Is the ability to disable the random seeding really needed though ? If we do use it, we should prefix it with "MM_".

I suggested to do this. If something breaks either during tests or in running a load-test we should be able to use the same seed when debugging. So this is just to help us really, nothing the user should be concerned with. Using `MM_` prefix sounds good.

**Comment by agnivade** [comment]
Ok, I am still okay with running the load test with a given seed. But adding main_test.go everywhere seems like a bit of an overkill to me. If a test fails, one can just manually set the seed themselves to debug.

**Comment by streamer45** [comment]
> Ok, I am still okay with running the load test with a given seed. But adding main_test.go everywhere seems like a bit of an overkill to me. If a test fails, one can just manually set the seed themselves to debug.

But how do we set a global seed shared by all tests? That was the problem we are trying to solve with `TestMain`. 
I couldn't find a way to do it, like a `init` function for everything.

**Comment by isacikgoz** [comment]
> I thought we wanted to do the same while running the test too ?

@agnivade I missed that one, thanks for reminding!

File: cmd/loadtest/loadtest.go [hunk 1]
```diff
@@ -30,6 +33,20 @@ func RunLoadTestCmdF(cmd *cobra.Command, args []string) error {
 
 	controllerType := config.UserControllerConfiguration.Type
 
+	s := os.Getenv("MM_LOADTEST_SEED")
+	var seed int64
+	if s != "" {
+		v, err := strconv.Atoi(s)
+		if err != nil {
+			mlog.Error(fmt.Sprintf("could not convert %q to a numeric value", s))
```
Comment (streamer45) [inline cmd/loadtest/loadtest.go]: I think we should exit the program with error if this happens. No need to continue.

File: loadtest/control/utils_test.go [hunk 2]
```diff
@@ -6,14 +6,35 @@ package control
...
 
+func TestMain(m *testing.M) {
+	s := os.Getenv("MM_LOADTEST_SEED")
+	var seed int64
+	if s != "" {
+		v, err := strconv.Atoi(s)
+		if err != nil {
+			panic(err)
+		}
+		seed = int64(v)
```
Comment (streamer45) [inline loadtest/control/utils_test.go]: This is one case where a *shared* util function would come in handy.

File: cmd/loadtest/loadtest.go (hunk 1 above)
Comment (isacikgoz) [reply cmd/loadtest/loadtest.go]: Was not so sure about this. Anyways, it's done 👍

File: loadtest/control/utils_test.go (hunk 2 above)
Comment (isacikgoz) [reply loadtest/control/utils_test.go]: Moved it 👍

**Pull request merged**